### DISCUSSION TITLE
Delaunay_2d function create mesh wiht loops line has inward angle.

### DISCUSSION THREAD

**Discussion by zhang9song** [pyvista/pyvista#4101] [Q&A]
Hi, guys

I have a loop lines has inward angle and want to use delaunay_2d function to create a mesh. The result is not correct. How to get right mesh  like picture?

```
import pyvista as pv
import numpy as np

points =np.array([[0, 0, 0], [2, 0, 0], [2, 1, 0], [1, 1, 0], [1, 2, 0], [0, 2, 0], [0,0,0]])
points_ploy = pv.PolyData(points)
lines = pv.lines_from_points(points)
filled_surface = points_ploy.delaunay_2d(edge_source=lines)
lines.plot(show_edges=True)
```

![image](https://user-images.githubusercontent.com/24886269/223668544-5cf24f20-d57b-431f-88d7-87293dc4f623.png)

![image](https://user-images.githubusercontent.com/24886269/223668374-98752c85-9d53-4d47-b1a2-ba7cfc1895bf.png)

**Comment by banesullivan** [comment]
This answer may help: https://github.com/pyvista/pyvista/discussions/3904#discussioncomment-4824064
